third_party/terraform/resources/resource_compute_instance.go.erb

```diff
@@ -1461,28 +1482,99 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 				if opErr != nil {
 					return opErr
 				}
-				// re-read fingerprint
-				instance, err = config.clientComputeBeta.Instances.Get(project, zone, instance.Name).Do()
+			}
+			return nil
+		}
+
+		if !updateDuringStop {
+			if d.HasChange(prefix + ".access_config") {
+				// Access configs cannot be updated, they must first be removed
+				if d.HasChange(prefix + ".access_config") {
+					err := deleteAccessConfigs()
+					if err != nil {
+						return err
+					}
+					err = createAccessConfigs()
+					if err != nil {
+						return err
+					}
+					err = readFingerPrint()
+					if err != nil {
+						return err
+					}
+				}
+			}
+			if d.HasChange(prefix + ".alias_ip_range") {
+				// Lets be explicit about what we are changing in the patch call
+				networkInterfacePatchObj := &computeBeta.NetworkInterface{}
+
+				// Alias IP ranges cannot be updated; they must be removed and then added
+				// unless you are changing subnetwork/network
+				if d.HasChange(prefix + ".alias_ip_range") {
+					err := deleteAliasIPRanges()
+					if err != nil {
+						return err
+					}
+					networkInterfacePatchObj.AliasIpRanges = networkInterface.AliasIpRanges
+					networkInterfacePatchObj.Fingerprint = instNetworkInterface.Fingerprint
```

Comment:
are we tracking too many variables at this point? why are the alias ip ranges and fingerprints getting set from two different network interfaces?